packages/rum-core/src/browser/performanceCollection.ts

```diff
@@ -154,6 +161,17 @@ export function startPerformanceCollection(lifeCycle: LifeCycle, configuration:
   }
 }
 
+function waitForRequestIdleOrUnload(lifeCycle: LifeCycle, callback: () => void) {
+  let hasBeenCalled = false
+  const callOnce = () => {
+    if (hasBeenCalled) return
+    hasBeenCalled = true
+    callback()
+  }
+  requestIdleCallback(callOnce)
+  lifeCycle.subscribe(LifeCycleEventType.BEFORE_UNLOAD, callOnce)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏what is the reasoning for calling it on unload?

Reply:
I don't know the detail behind the idle period detection but, FMU we can have cases where they are no idle periods before the unload. This is why the requestIdleCallback API provides a timeout option. 

Btw the documentation states: "A timeout option is strongly recommended for required work, as otherwise it's possible multiple seconds will elapse before the callback is fired." but in our case BEFORE_UNLOAD seems more appropriate.

Reply:
IMO, these kind of behaviors should be explained in the code and tested. 
I am wondering if this processing is not likely to be aborted by the unload anyway.

On the other hand, if we are worried about the requestIdleCallback not being called due to too much activity, let's just use a setTimeout, it should at least split this processing from the init.

wdyt?

Reply:
Other concern about the async call:
https://github.com/DataDog/browser-sdk/pull/1337#discussion_r808080557